backwardcompatibilityml/helpers/training.py

```diff
@@ -504,36 +513,33 @@ def evaluate_model_performance_and_compatibility_on_dataset(h1, h2, dataset, per
         the trust compatibility score of h2 with respect to h1, and
         the error compatibility score of h2 with respect to h1.
     """
-    number_of_batches = len(dataset)
     h1_dataset_error_instance_ids = []
     h2_dataset_error_instance_ids = []
     h1_and_h2_dataset_error_instance_ids = []
-    h2_dataset_error_instance_ids_by_class = {}
+    h1h2_dataset_incompatible_instance_ids_by_class = {}
     classes = set()
     all_error_instances = []
     for batch_ids, data, target in dataset:
         classes = classes.union(target.tolist())
         h1_error_count_batch, h2_error_count_batch, h1_and_h2_error_count_batch =\
             get_model_error_overlap(h1, h2, batch_ids, data, target, device=device)
-        h2_error_instance_ids_by_class =\
-            get_error_instance_ids_by_class(h2, batch_ids, data, target, device=device)
-        all_errors = get_all_error_instance_indices(
-            h1, h2, data, target, device=device)
+        (all_errors, h1h2_incompatible_ids_by_class) = get_all_error_instance_indices(
```

Comment:
So I would do the following:

1. Use the response from the un-changed `get_all_error_instance_indices` here, to calculate the incompatible points by class.

2. Probably add another function to isolate that calculation and make a call to it here.